### DISCUSSION TITLE
"Update with rebase" results in a large change set with spurious commits

### DISCUSSION THREAD

**Discussion by alalazo** [community/community#51948] [Pull Requests] [answered]
### Select Topic Area

Bug

### Body

I am a maintainer and core developer of the Spack project https://github.com/spack/spack Today, it happened on 2 different pull requests in that project to click the "Update with rebase" button and get the UI display a large changeset. For instance on this PR https://github.com/spack/spack/pull/36329 ( a one line change) I obtained the following:

![Screenshot from 2023-04-04 14-18-48](https://user-images.githubusercontent.com/4199709/229800525-bf1785da-922f-4334-b9d3-b6e651f7c6f2.png)

The funny thing is that when I checked the 231 commits listed, they were all replicating our target branch - except the last one which was supposed to be the PR. To be clearer, when this happened the HEAD of the `develop` branch was [6aa4c29119ea241d2370d3e3ce6e35f76001219b](https://github.com/spack/spack/commit/6aa4c29119ea241d2370d3e3ce6e35f76001219b) and so was `HEAD~1` of the PR branch.[^1]

I fixed the issue in this PR by force pushing to the user branch, but was wondering if this has been reported already, or if it is an issue on my side that somehow started today.

[^1]: It feels like the pull request updated the branch, but was still trying to merge with the old `develop` commit. I didn't verify this hypothesis though.

**Comment by keith** [comment]
We started seeing this yesterday as well. I just went to try and reproduce to file a bug report and it seems that I can't anymore

**Reply by keith** [reply]
I was wrong, we're still seeing it. Closing and reopening the PR also fixes the issue

**Comment by kadru** [comment]
I have the same issue with differents PRs, but is in a private repo.

**Comment by philippeauriach** [comment]
We have this issue as well on private repositories. It seems to be only a graphical issue, as merging result in a correct tree in master.

We found a workaround, to close and reopen the PR result in the proper list of commits being displayed.

**Comment by sjaeckel** [comment]
Same here, fixed it by amending the most recent commit on the source-branch and force-pushing it.

**Comment by ikex** [comment]
It's not just a graphical issue - it also affects other PR sections like "Files changed" (it shows difference in commits against the... snapshot main from the time when the PR has been created... it renders this part kinda cumbersome) and - what's even worse - triggers all pipelines affected by imported commits. So - for instance - if we "rebase" commits that contain changes to specific paths it will trigger those additional pipelines. Or - for instance - we have a specific case for dependabot PRs where commit linter treats those commits differently... if we pull those commits by "rebase" it will trigger a pipeline that will fail on commitlint, just because the PR is not by dependabot.

So i think it's not that minor issue and it affects us badly.

Currently we are rebasing locally and pushing changes to PRs.

**Comment by DarylCantrell** [comment, marked as answer]
GitHub staff here. Thank you for the bug report. This was caused by a new feature we are rolling out. We have deactivated it for the moment, and a permanent fix for this issue is in the merge queue. It is safe to rebase again.

If you have existing pull requests with redundant commits, it should suffice to close and then re-open the pull request. This causes the system to recompute the PR base commit and it should then display correctly.

Sorry for the trouble, and thanks for letting us know.

**Reply by pierresisson** [reply]
NOICE

**Reply by nhtyy** [reply]
I still get this frequently

**Reply by prathamesh-NS1** [reply]
still not fixed

**Comment by nguyta02** [comment]
We are facing an issue similar to this, but not sure.
@DarylCantrell  - Has this been fixed ?

**Reply by nhtyy** [reply]
+1

**Reply by nhtyy** [reply]
Usually I see it when I update a base branch with a large amount of commits, then rebase a feature branch on top of that

**Comment by sapbusevarkan** [comment]
I had the same issue this morning and I think it is not solved yet maybe. You do not need to close and re-open the PR, just do this:
merge xx commits into [main] from [your branch]
here select another branch and then re-select main. it will be fixed and you will only see your changes.

at least it worked for me :)

**Comment by drod3763** [comment]
This is still happening when using Update with Rebase.
